src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/IIdGeneratorService.java

```diff
@@ -0,0 +1,7 @@
+package uk.gov.companieshouse.api.strikeoffobjections.service;
+
+import java.security.NoSuchAlgorithmException;
+
+public interface IIdGeneratorService {
+    String generateId() throws NoSuchAlgorithmException;
```

Comment:
I can't see where the declared exception is thrown or declared in the actual implementing method. Is it required?